jiant/proj/main/modeling/taskmodels.py

```diff
@@ -120,6 +120,28 @@ def forward(self, batch, task, tokenizer, compute_loss: bool = False):
             return LogitsOutput(logits=logits, other=encoder_output.other)
 
 
+class SpanPredictionModel(Taskmodel):
+    def __init__(self, encoder, span_prediction_head: heads.TokenClassificationHead):
+        super().__init__(encoder=encoder)
+        self.offset_margin = 1000
```

Comment:
What's the significance of 1,000? Drop a comment in the code to explain?

Reply:
1000 is a big enough number that exp(-1000) will be strict 0 in float32. so that if we add 1000 to the legit dimensions in the input of softmax, we guarantee the output distribution will only be non-zero at those dimensions.